utils/build-script-impl

```diff
@@ -2263,6 +2264,15 @@ for host in "${ALL_HOSTS[@]}"; do
 done
 # END OF BUILD PHASE
 
+# Deploy Android built products. This is done prior to the install phase, since
+# Android host tests won't run correctly unless the built stdlib is pushed to
+# the device running the tests.
+if [[ ! -z "${ANDROID_DEPLOY_DEVICE_PATH}" ]]; then
+    echo "--- Deploying built products to Android device ---"
+    set -x
+    "${SWIFT_SOURCE_DIR}"/utils/android/adb_push_built_products.py --ndk "${ANDROID_NDK}" --destination "${ANDROID_DEPLOY_DEVICE_PATH}" "$(build_directory ${HOST_TARGET} swift)"/lib/swift/android/
+    { set +x; } 2>/dev/null
+fi
```

Comment:
Uploading should be done in `test/CMakeLists.txt` -- just set the `command_upload_stdlib` if `SDK` is Android.  This way, just doing `ninja check-swift-android` will do the right thing and use the latest library.